tests/publiccloud/run_ltp.pm

```diff
@@ -37,20 +25,16 @@ sub run {
     my $provider = $self->{provider} = $self->provider_factory();
     $provider->init();
 
-    my $instance = $provider->ipa(
-        instance_type => get_var('PUBLIC_CLOUD_INSTANCE_TYPE'),
-        cleanup       => 0,
-        image_id      => $self->get_image_id($provider)
-    );
+    my $instance = $provider->create_instance(image => $self->get_image_id($provider));
```

Comment:
I think that it would be only case how we will call `create_instance` meaning it will be always  `$self->get_image_id($provider)` which tells me that we can call it inside

Reply:
Yes, we can make this parameter optional.

Reply:
Not sure how to do this since $provider needs to be passed to provider.pm to be able to call get_image_id()

Reply:
$provider is created from provider_factory() in basetest.pm

https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/6234/files#diff-8eff9f64c4a8410248c1ee15006eb96cL37

in provider.pm, $self is an object of the type 'publiccloud::gce'

Reply:
I had the idea,to not use the openQA variables inside provider classes.
What you could do is call the function directly

```
publiccloud::basetest::get_image_id($self)
```

But I'm wondering that it work in general. There seems to be an error in `publiccloud::basetest::get_image_id()` cause the first parameter if it is called like $self->get_image_id(), it is the `publiccloud::basetest` instead of the `publiccloud::provider`.

Reply:
I guess it only worked, cause we always had the upload_img job in all places where PUBLIC_CLOUD_IMAGE_ID was set.